enhancements/opm-content-management.md

```diff
@@ -0,0 +1,145 @@
+---
+title: opm-content-management 
+authors:
+  - "@anik120"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+status: implementable
+---
+
+# OPM Content Management
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Summary
+
+This enhancement is a proposal for extending the existing capabilites of the `opm` command line utility, to provide more fine grained control over the management of catalog indexes.   
+
+`opm` will have significant improvement to UX for editing an index (removing individual bundles from a package, removing channels from a package). Indexes can also be queried directly for inspection, and update graph visualization for individual packages will be added. Bundle validation without a deamon will is also proposed in this enhancement. 
+
+## Motivation
+
+As more and more operators are migrated to the new bundle format, and more index images are created for Operator Catalogs, creators of catalogs of operators, such as the Red Hat pipeline responsible for maintaining the `redhat-operators` catalog, should have more fine grained control over the management of the catalogs using the `opm` tooling.  
+
+### Goals
+
+- Allow `opm` users to perform bundle validations in environments without a Docker daemon. Bundle validation also includes validation of bundle image labels against `metadata/annotations.yaml`
+- Allow `opm` users to safely edit indexes, which includes removing bundle version from a package in an index, and removing a channel from a bundle. 
```

Comment:
IIRC we also want to edit the channels

Reply:
example: promotion from techpreview to stable

Reply:
@gallettilance do you mean change the default channel of a package?

Reply:
any channel

Reply:
this needs to handle deprecation (say the 4.1 channel should be removed) and promotion  (say the techpreview becomes stable)

Reply:
@gallettilance this section tackles the deprecation of a channel: https://github.com/operator-framework/enhancements/pull/26/files#diff-e5605260b53989d798d8558d3bf4c1f0R93

However, promotion of techpreview to stable seems like creating a discrepancy between what the operator author had communicated (via CSVs) and what the index author is attempting to do.

Reply:
> promotion of techpreview to stable seems like creating a discrepancy between what the operator author had communicated (via CSVs) and what the index author is attempting to do.

I think we need to treat what the CSV says as the initial state, not the final state. The workflow for channel promotion becomes hairy (and you lose provenance - the bundle digest will change after promotion) if we don't allow the index to be authoritative on this